zerver/models.py

```diff
@@ -3069,19 +3069,33 @@ def is_status_message(content: str, rendered_content: str) -> bool:
     class Meta:
         indexes = [
             GinIndex("search_tsvector", fastupdate=False, name="zerver_message_search_tsvector"),
-            models.Index(Upper("subject"), name="upper_subject_idx"),
-            models.Index("date_sent", name="zerver_message_date_sent_3b5b05d8"),
             models.Index(
-                "recipient",
+                "realm_id", "recipient_id", "id", name="zerver_message_realm_recipient_id"
+            ),
+            models.Index(
+                "realm_id",
+                "sender_id",
+                "recipient_id",
+                name="zerver_message_realm_sender_recipient",
+            ),
+            models.Index(
+                "realm_id", "sending_client_id", name="zerver_message_realm_sending_client"
+            ),
+            models.Index("realm_id", Upper("subject"), name="zerver_message_realm_upper_subject"),
```

Comment:
I can't think of a query where we'd use this index without having the `recipient_id` in the query... oh, I guess you can search for `topic:foo` in search, and we need that to not be a table scan. Fine, we do need this for that.